pkg/services/hcloud/server/server_test.go

```diff
@@ -339,3 +341,99 @@ var _ = Describe("Test ValidateLabels", func() {
 		}),
 	)
 })
+
+var _ = Describe("Test handleRateLimit", func() {
+	type testCaseHandleRateLimit struct {
+		hm           *infrav1.HCloudMachine
+		err          error
+		functionName string
+		errMsg       string
+		expectError  error
+	}
+
+	DescribeTable("Test handleRateLimit",
+		func(tc testCaseHandleRateLimit) {
+			err := handleRateLimit(tc.hm, tc.err, tc.functionName, tc.errMsg)
+			if tc.expectError != nil && hcloud.IsError(err, hcloud.ErrorCodeRateLimitExceeded) {
+				Expect(err).To(MatchError(tc.expectError))
+				Expect(isPresentAndFalseWithReason(tc.hm, infrav1.HetznerAPIReachableCondition, infrav1.RateLimitExceededReason)).To(BeTrue())
```

Comment:
can you just add a property to your test case that says "condition set" on true or false? in the one case you expect it to be set, in the other one you expect it to NOT be set